debezium-ai/debezium-ai-embeddings-base/src/main/java/io/debezium/ai/embeddings/AbstractEmbeddingsSmt.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.ai.embeddings;
+
+import static java.lang.String.format;
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.components.Versioned;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.Module;
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.data.vector.FloatVector;
+import io.debezium.transforms.ConnectRecordUtil;
+import io.debezium.transforms.SmtManager;
+import io.debezium.util.BoundedConcurrentHashMap;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+
+/**
+ * Base class of the embeddings SMT, which provides common configuration and manipulations with the Connect record.
+ * Child classes has to implement at least {@link AbstractEmbeddingsSmt#getModel()} class, which provides specific
+ * model which will be used for creating the embeddings.
+ *
+ * @author vjuranek
+ */
+public abstract class AbstractEmbeddingsSmt<R extends ConnectRecord<R>> implements Transformation<R>, Versioned {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractEmbeddingsSmt.class);
+
+    private static final Field TEXT_FIELD = Field.create("field.source")
+            .withDisplayName("Name of the record field from which embeddings should be created.")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .required()
+            .withDescription("Name of the field of the record which content will be used as an input for embeddings. Supports also nested fields.");
+
+    private static final Field EMBEDDGINS_FIELD = Field.create("field.embeddings")
```

Comment:
Can we make it optional and replace the whole payload with embeddings hwn not provided?

Reply:
sure